apps/challenges/views.py

```diff
@@ -42,6 +42,12 @@ def challenge_list(request, challenge_host_team_pk):
         return paginator.get_paginated_response(response_data)
 
     elif request.method == 'POST':
+
+        if challenge_host_team.created_by != request.user:
```

Comment:
@gauthamzz I was thinking deeply about this and it seems like this won't cover all the use cases. For example, if a host team consists of 4 members then only the member who has created the team can only create a challenge using that team but ideally what should happen is that all of the members of that team should have the right to create a team? Does that makes sense? Let me know if you have concerns.